src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,241 @@
+                var fullProgId = vsProgId.StartsWith("!") ? vsProgId : $"!{vsProgId}";
+
+                if (displayName.Equals(fullProgId, StringComparison.OrdinalIgnoreCase) &&
+                    (displayNameProcessId == process.Id))
+                {
+                    runningObjectTable.GetObject(moniker, out dte);
+                }
+            }
+            while (dte == null);
+
+            return (DTE)(dte);
+        }
+
+        public static async Task<string> LocateFile(string fileName, string downloadUrl = "")
+        {
+            if (File.Exists(fileName))
+            {
+                return Path.GetFullPath(fileName);
+            }
+
+            var filePath = Path.Combine(Environment.CurrentDirectory, fileName);
+
+            if (File.Exists(filePath))
+            {
+                return filePath;
+            }
+
+            filePath = Path.Combine(typeof(IntegrationHelper).Assembly.Location, fileName);
+
+            if (File.Exists(filePath))
+            {
+                return filePath;
+            }
+
+            if (!string.IsNullOrWhiteSpace(downloadUrl))
+            {
+                await DownloadFileAsync(downloadUrl, fileName).ConfigureAwait(continueOnCapturedContext: false);
+            }
+
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException($"Unable to locate the specified file. (FileName: '{fileName}')", fileName);
+            }
+            
+            return filePath;
+        }
+
+        public static object GetRegistryKeyValue(RegistryKey baseKey, string subKeyName, string valueName)
+        {
+            using (var registryKey = baseKey.OpenSubKey(subKeyName))
+            {
+                if (registryKey == null)
+                {
+                    throw new Exception($"The specified registry key could not be found. Registry Key: '{registryKey}'");
+                }
+
+                return registryKey.GetValue(valueName);
+            }
+        }
+
+        public static bool TryDeleteDirectoryRecursively(string path)
+        {
+            try
+            {
+                DeleteDirectoryRecursively(path);
+                return true;
+            }
+            catch (Exception e)
+            {
+                IntegrationLog.Current.Warning($"Failed to recursively delete the specified directory. (Name: '{path}')");
+                IntegrationLog.Current.WriteLine($"\t{e}");
+                return false;
+            }
+        }
+
+        public static async Task WaitForResultAsync<T>(Func<T> action, T expectedResult)
+        {
+            while (!action().Equals(expectedResult))
+            {
+                await Task.Yield();
+            }
+        }
+
+        public static async Task<T> WaitForNotNullAsync<T>(Func<T> action) where T : class
+        {
+            var result = action();
```

Comment:
because this one returns the non null result, while the other just waits for a condition to become true.